### ISSUE TITLE
autotype does not type umlauts

### ISSUE THREAD

**Issue by elandorr** [keepassxreboot/keepassxc#9650]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. have an umlaut in a password
2. use auto-type
3. wonder why it fails
4. try it on a visible text field and see it omitted the umlauts, possibly more characters

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.4
Revision: 63b2394

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux
Desktop Env: none 
Windowing System: X11

**Comment by droidmonkey** [comment]
See #7881.

**Comment by elandorr** [comment]
@droidmonkey For the record, there was no dead key involved, and the keyboard layout used is standard `us`.

**Comment by droidmonkey** [comment]
Standard US does not support umlats without a complex key sequence. Whenever you release your hands from the keyboard before typing the next letter, that is an example of a dead key type action, in this case called "compose key".

**Comment by elandorr** [comment]
I know what a deadkey is, but thanks for explaining why this applies here. Is there a specific reason you emulate a complex sequence vs. using the clipboard? Or is that too annoying due to the clipboard idiosyncrasies of the last decades?

The original keepass on windows has an obfuscation where it jumps around and mixes it up. Since we don't have that in XC, wouldn't malware be able to read either just the same?

**Comment by droidmonkey** [comment]
Obfuscation of typing is worthless, it is discussed in the closed issues (TCATO is the acronym). Using the clipboard is the exact opposite of what we want when using Auto-Type, you using keyboard actions specifically to avoid using the clipboard. 

For your use case, we recommend either removing the complex characters from the password or using the clipboard instead of Auto-Type. 

P.S. - if you have malware already running on your system then you have lost. There is absolutely nothing KeePassXC or any other program can do to prevent theft of information in use.

**Comment by elandorr** [comment]
> Using the clipboard is the exact opposite of what we want when using Auto-Type

What *is* the point of autotype in your book? I use it to not have to manually paste.

What I meant is what happens when you start autotype is irrelevant to the user, unless I'm missing something:

- send keystrokes that emulate the needed, possibly complex sequence that nobody ever uses, possibly depends on locale trickery (sounds like a nightmare)
 
vs.

- save a character to clipboard, paste it, repeat

vs.

- save whole string to clipboard, paste it

So I figured you chose the first for security reasons.

> P.S. - if you have malware already running on your system then you have lost. 

Sure, but as usual this is a gradient. Windows has a separate ['safe desktop'](https://security.stackexchange.com/questions/3759/how-does-the-windows-secure-desktop-mode-work#3762) where passwords can be entered that may thwart some sniffing. (veracrypt, keepass use that)

The classic rats most of us have seen at some point are full auto, if you have a bunch of gibberish in between, it makes it annoying. Not 'secure' by any means, but 9/10 times if there's an easier victim they won't waste time on you. I see why that'd be 'nice-to-have' on windows, with mass malware, not specific targets.

> we recommend either removing the complex characters from the password or using the clipboard instead of Auto-Type.

Yup, been pasting manually ever since this started bugging. But only recently I noticed the character issue. Removing the complex chars is not an option, as many sites enforce short passwords. I'd surely prefer long plain ascii that we can type manually worst-case.
